packages/allure-cypress/src/browser/patching.ts

```diff
@@ -0,0 +1,130 @@
+import { isPromise } from "allure-js-commons/sdk";
+import type { CypressSuiteFunction, DirectHookImplementation, HookImplementation } from "../types.js";
+import {
+  completeSpecIfNoAfterHookLeft,
+  completeSpecOnAfterHookFailure,
+  throwAfterSpecCompletion,
+} from "./lifecycle.js";
+
+/**
+ * Patches the `after` function, to inject reporting of spec-level
+ * `after` hooks defined by the user.
+ */
+export const enableScopeLevelAfterHookReporting = () => {
+  const [getSuiteDepth, incSuiteDepth, decSuiteDepth] = createSuiteDepthCounterState();
+  patchDescribe(incSuiteDepth, decSuiteDepth);
+  patchAfter(getSuiteDepth);
+};
+
+const createSuiteDepthCounterState = (): [get: () => number, inc: () => void, dec: () => void] => {
```

Comment:
Guess, it would be much more convenient to work with object, not an array here because we need to keep in mind an order of the functions to use them properly, but in the case of an object, it's stupidly simple:

```js
const { count, increment, decrement } = createSuiteDepthCounterState()

// ...
```

Reply:
Nice suggestion, done